From 97ee6ca5811e9b68c5c7a3a52b24677a301bf78b Mon Sep 17 00:00:00 2001
From: Thomas Sibley <trs@bestpractical.com>
Date: Wed, 27 Feb 2013 17:19:22 -0800
Subject: [PATCH] Add a callback for modifying the displayed queues on the
 quicksearch prefs

This also lets the callback skip the ShowTicket rights check or specify
a different right.
---
 share/html/Prefs/Quicksearch.html | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/share/html/Prefs/Quicksearch.html b/share/html/Prefs/Quicksearch.html
index 02b88ab..93b026f 100644
--- a/share/html/Prefs/Quicksearch.html
+++ b/share/html/Prefs/Quicksearch.html
@@ -71,7 +71,16 @@ my $user = $session{'CurrentUser'}->UserObj;
 my $unwanted = $user->Preferences('QuickSearch', {});
 my $Queues = RT::Queues->new($session{'CurrentUser'});
 $Queues->UnLimit;
-my @queues = grep {$_->CurrentUserHasRight('ShowTicket')} @{$Queues->ItemsArrayRef};
+
+my $right = 'ShowTicket';
+$m->callback(
+    CallbackName    => 'ModifyQueues',
+    Queues          => \$Queues,
+    Right           => \$right,
+    Unwanted        => $unwanted,
+);
+
+my @queues = grep { $right ? $_->CurrentUserHasRight($right) : 1 } @{$Queues->ItemsArrayRef};
 
 if ($ARGS{'Save'}) {
     for my $queue (@queues) {
-- 
1.7.11.3

